### PR TITLE
Fix (inverse) meaning of -persistmempool

### PR DESCRIPTION
**Pull request by maflcko**
Passing `-persistmempool` is currently treated as `-nopersistmempool`

### CODE REVIEW

**Comment by laanwj** [comment]
Any idea when this regressed, or has this been the case since the option was introduced?

**Comment by MarcoFalke** [comment]
I haven't checked, but I remember seeing this bug (#23062) since I started working on Bitcoin Core, though I might be mistaken.

**Comment by laanwj** [comment]
Ok adding backport label for 0.21.x as well then. That seems far enough back.

**Review by ryanofsky — APPROVED** [review]
I think change of behavior for `-persistmempool` should be noted in release notes, but otherwise code review ACK faff17bbde6dcb1482a6210bc48b3192603a446f

**Comment by jnewbery** [comment]
Tested ACK faff17bbde6dcb1482a6210bc48b3192603a446f

Verified that cherry-picking just the test change fails on master.

> Any idea when this regressed, or has this been the case since the option was introduced?

Probably since it was introduced in #9966. Fortunately, I think it's unlikely that anyone would be using `-persistmempool` since the default is true.

**Comment by MarcoFalke** [comment]
> it's unlikely that anyone would be using -persistmempool since the default is true.

Indeed. Also, there is no bug report. So maybe we can skip the backport, as the bugfix is also a (silent) behaviour change.

File: doc/release-notes.md [hunk 1]
```diff
@@ -86,6 +86,12 @@ New settings
 Updated settings
 ----------------
 
+- The meaning of the command line option `-persistmempool` (without a value
+  provided) incorrectly disabled mempool persistance. `-persistmempool` is now
+  treated like other boolen options to mean `-persistmempool=1`.
```
Comment (jonatack) [inline doc/release-notes.md:91]: nit, s/boolen/boolean/

File: doc/release-notes.md [hunk 2]
```diff
@@ -86,6 +86,12 @@ New settings
...
+  `-persistmempool=0`, `-persistmempool=1` and `-nopersistmempool` are
```
Comment (jonatack) [inline doc/release-notes.md:92]:
```suggestion
  Passing `-persistmempool=0`, `-persistmempool=1`, and `-nopersistmempool` is
```

**Review by jonatack — COMMENTED** [review]
Nice find.

File: doc/release-notes.md (hunk 1 above)
Comment (blockskillally) [reply doc/release-notes.md:91]: Why? 🦆👀🔐🆔🔐⭕

File: doc/release-notes.md (hunk 2 above)
Comment (blockskillally) [reply doc/release-notes.md:92]: - [ ] Why. []()

File: doc/release-notes.md (hunk 1 above)
Comment (blockskillally) [reply doc/release-notes.md:91]: CNSatoshisPerk>0 )? Diff-- git>(1)=earned. h self. Genate (self. Nodes (1)><(I).2 self. Sync_all(ii)@@-179.7+149.7@@ diff run?

File: doc/release-notes.md (hunk 2 above)
Comment (blockskillally) [reply doc/release-notes.md:92]: - [ ] Tothemoon💯🐎📨🆔⭕🦆

Comment (blockskillally) [reply doc/release-notes.md:92]: But it's the opposite always isn't it?

File: doc/release-notes.md (hunk 1 above)
Comment (MarcoFalke) [reply doc/release-notes.md]: Done

File: doc/release-notes.md (hunk 2 above)
Comment (MarcoFalke) [reply doc/release-notes.md:92]: Done

**Comment by jnewbery** [comment]
ACK fa94851fcd

**Review by ryanofsky — APPROVED** [review]
Code review ACK fa94851fcd1636549c35bc7c3b42348a350d5741. (I thought I acked this previously, but my comments were stuck in draft. Feel free to ignore suggestions below which may be a little late)

File: doc/release-notes.md [hunk 3]
```diff
@@ -86,6 +86,12 @@ New settings
 Updated settings
 ----------------
 
+- The meaning of the command line option `-persistmempool` (without a value
+  provided) incorrectly disabled mempool persistance. `-persistmempool` is now
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply doc/release-notes.md]: Thanks, done

File: doc/release-notes.md [hunk 4]
```diff
@@ -86,6 +86,12 @@ New settings
 Updated settings
 ----------------
 
+- The meaning of the command line option `-persistmempool` (without a value
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply doc/release-notes.md]: Thanks, done

**Review by hebasto — APPROVED** [review]
ACK faa9c19a4b27e7fabf7c5deae1b4c4ca612ed01a, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by jnewbery** [comment]
reACK faa9c19a4b27e7fabf7c5deae1b4c4ca612ed01a

**Comment by MarcoFalke** [comment]
Removed from 0.21 backport for now, because this is a behaviour change. It is not affecting anyone with high likelyhood.

If it is backported to 22.0, the release notes can be cleared in `master`.

**Comment by fanquake** [comment]
Being backported to 22.x in #23276.

**Pull request merged**